### PR TITLE
Fix 1346, add missing taxonomy term labels trunk

### LINKED ISSUE #1346: Make sure that all taxonomies have all relevant labels when registered
Not all of the taxonomies created by Largo in inc/taxonomies.php have all labels that may be relevant: https://codex.wordpress.org/Function_Reference/register_taxonomy#Arguments

Compare the labels used to register the `post-type` taxonomy (used for icons): 

https://github.com/INN/Largo/blob/6be6de05dac472cda1be018747c64549e398280f/inc/taxonomies.php#L135-L145

``` php
                'labels' => array(
                    'name' => __( 'Post Types', 'largo' ),
                    'singular_name' => __( 'Post Type', 'largo' ),
                    'all_items' => __( 'All Post Types', 'largo' ),
                    'edit_item' => __( 'Edit Post Type', 'largo' ),
                    'update_item' => __( 'Update Post Type', 'largo' ),
                    'view_item' => __( 'View Post Type', 'largo' ),
                    'add_new_item' => __( 'Add New Post Type', 'largo' ),
                    'new_item_name' => __( 'New Post Type Name', 'largo' ),
                    'search_items' => __( 'Search Post Type'),
                ),
```

With the labels used for the tax landing pages: https://github.com/INN/Largo/blob/6be6de05dac472cda1be018747c64549e398280f/inc/wp-taxonomy-landing/functions/cftl-admin.php#L20-L34

``` php
        'labels' => array(
            'name' => __('Landing Pages', 'cf-tax-landing'),
            'singular_name' => __('Landing Page', 'cf-tax-landing'),
            'add_new' => _x('Add New', 'cftl-tax-landing', 'cf-tax-landing'),
            'add_new_item' => __('Add New Landing Page', 'cf-tax-landing'),
            'edit_item' => __('Edit Landing Page', 'cf-tax-landing'),
            'new_item' => __('New Landing Page', 'cf-tax-landing'),
            'all_items' => __('All Landing Pages', 'cf-tax-landing'),
            'view_item' => __('View Landing Page', 'cf-tax-landing'),
            'search_items' => __('Search Landing Pages', 'cf-tax-landing'),
            'not_found' =>  __('No Landing Pages found', 'cf-tax-landing'),
            'not_found_in_trash' => __('No Landing Pages found in Trash', 'cf-tax-landing'),
            'parent_item_colon' => '',
            'menu_name' => __('Landing Pages', 'cf-tax-landing')
        ),
```

See #256 for when all those labels were added for landing pages.

### PR DESCRIPTION
**Pull request by charmoney**
## Changes

This pull request makes the following changes:

- Add missing taxonomy labels for prominence, series, and post-type. Fixes #1346

## Why

For #1346, referred as related by #1776

## Testing/Questions

Features that this PR affects:

- Custom Taxonomies

Questions that need to be answered before merging:

- [x] Is this PR targeting the correct branch in this repository?

Steps to test this PR:

1. Check out branch
1. `wp taxonomy get --field=labels prominence`
1. Verify labels do *not* include the default category or tag labels for any key.
1. `wp taxonomy get --field=labels post-type`
1. Verify labels do *not* include the default category or tag labels for any key.
1. `wp taxonomy get --field=labels series`
1. Verify labels do *not* include the default category or tag labels for any key.

## Additional information

INN Member/Labs Client requesting: (if applicable)

- [x] Contributor has read INN's [GitHub code of conduct](https://github.com/INN/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] Contributor would like to be mentioned in the release notes as: Chris Harmoney (Cornershop Creative)
- [x] Contributor agrees to the license terms of this repository.

### CODE REVIEW

**Comment by joshdarby** [comment]
@benlk I think this looks good. All 3 custom taxonomies are updated to include the same labels and they all work as expected.

File: inc/taxonomies.php [hunk 1]
```diff
@@ -134,15 +139,22 @@ function largo_custom_taxonomies() {
...
-					'singular_name' => __( 'Post Type', 'largo' ),
-					'all_items' => __( 'All Post Types', 'largo' ),
-					'edit_item' => __( 'Edit Post Type', 'largo' ),
-					'update_item' => __( 'Update Post Type', 'largo' ),
-					'view_item' => __( 'View Post Type', 'largo' ),
-					'add_new_item' => __( 'Add New Post Type', 'largo' ),
-					'new_item_name' => __( 'New Post Type Name', 'largo' ),
-					'search_items' => __( 'Search Post Type'),
+					'name'                  => __( 'Post Types', 'largo' ),
+					'singular_name'         => __( 'Post Type', 'largo' ),
```
Comment (joshdarby) [inline inc/taxonomies.php]: @benlk Do we want `name` and `singular_name` here to allow for specifying a context for translation like in the `Post Prominence` and `Series` taxonomies?

Comment (benlk) [reply inc/taxonomies.php]: yes, probably something like "taxonomy label".

File: inc/taxonomies.php [hunk 2]
```diff
@@ -46,18 +46,23 @@ function largo_custom_taxonomies() {
...
+					'view_item'             => __( 'View Post Prominence' ),
+					'update_item'           => __( 'Update Post Prominence' ),
+					'add_new_item'          => __( 'Add New Post Prominence' ),
+					'new_item_name'         => __( 'New Post Prominence Name' ),
+					'menu_name'             => __( 'Post Prominence' ),
+					'not_found'             => __( 'No post prominences found.' ),
+					'no_terms'              => __( 'No post prominences' ),
+					'items_list_navigation' => __( 'Post Prominences list navigation' ),
+					'items_list'            => __( 'Post Prominences list' ),
+					'back_to_items'         => __( '&larr; Back to Post Prominences' ),
```
Comment (benlk) [inline inc/taxonomies.php]: These all need the text domain `'largo'` to be specified.

**Pull request merged**